src/Agent.Worker/Handlers/PowerShell3Handler.cs

```diff
@@ -46,8 +46,17 @@ public async Task RunAsync()
             ArgUtil.File(moduleFile, nameof(moduleFile));
 
             // Craft the args to pass to PowerShell.exe.
+            /* 
+            Updates from Tim Brigham
+            This used to be a whole block of powershell passed on the CLI, but has been simplified into a wrapper "Use-VstsTaskScript.ps1"
+            This wrapper contains the original logic such as checking encoding settings, pshome variable and import statements.
+            However since it is now a file on disk it can be signed and pass WDAC / CLM requirements prior to executing Invoke-VstsTaskScript 
+            which is the original entry point for the task script.
+            The wrapper script is located in the bin\powershell folder of the agent installation. Since the process runs from the task directory
+            we need to move up and over to reach the wrapper script.
+            */ 
             string powerShellExeArgs = StringUtil.Format(
-                @"-NoLogo -Sta -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command "". ([scriptblock]::Create('if ([Console]::InputEncoding -is [Text.UTF8Encoding] -and [Console]::InputEncoding.GetPreamble().Length -ne 0) {{ [Console]::InputEncoding = New-Object Text.UTF8Encoding $false }} if (!$PSHOME) {{ $null = Get-Item -LiteralPath ''variable:PSHOME'' }} else {{ Import-Module -Name ([System.IO.Path]::Combine($PSHOME, ''Modules\Microsoft.PowerShell.Management\Microsoft.PowerShell.Management.psd1'')) ; Import-Module -Name ([System.IO.Path]::Combine($PSHOME, ''Modules\Microsoft.PowerShell.Utility\Microsoft.PowerShell.Utility.psd1'')) }}')) 2>&1 | ForEach-Object {{ Write-Verbose $_.Exception.Message -Verbose }} ; Import-Module -Name '{0}' -ArgumentList @{{ NonInteractive = $true }} -ErrorAction Stop ; $VerbosePreference = '{1}' ; $DebugPreference = '{1}' ; Invoke-VstsTaskScript -ScriptBlock ([scriptblock]::Create('. ''{2}'''))""",
+                @"-NoLogo -Sta -NoProfile -ExecutionPolicy Unrestricted -Command ""..\..\..\..\bin\powershell\Use-VstsTaskScript.ps1"" -Name {0} -DebugOption {1} -ScriptBlock ""{2}""",
```

Comment:
Please don't use relative paths.
I suggest to build path to script using constant and context, e.g. `Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Bin), "powershell", "Use-VstsTaskScript.ps1");`.
And please rename `Use-VstsTaskScript.ps1` => `Start-AzpTask.ps1` (me are moving away from the old VSTS brand)

Reply:
Naming convention is updated, and I have updated the script path to use the well known directory.